webv2/primarykey/primarykey_service.go

```diff
@@ -0,0 +1,155 @@
+func getColumnId(spannerTable ddl.CreateTable, columnName string) int {
+
+	var id int
+	for _, col := range spannerTable.ColDefs {
+		if col.Name == columnName {
+			id = col.Id
+		}
+	}
+	return id
+}
+
+// getSpannerTable return spannerTable for given TableId.
+func getSpannerTable(sessionState *session.SessionState, pkRequest PrimaryKeyRequest) (spannerTable ddl.CreateTable, found bool) {
+
+	for _, table := range sessionState.Conv.SpSchema {
+
+		if pkRequest.TableId == table.Id {
+			spannerTable = table
+			found = true
+		}
+	}
+	return spannerTable, found
+}
+
+// getColumnName return columnName for given columnId.
+func getColumnName(spannerTable ddl.CreateTable, columnId int) string {
+
+	var columnName string
+	for _, col := range spannerTable.ColDefs {
+		if col.Id == columnId {
+			columnName = col.Name
+		}
+	}
+	return columnName
+}
+
+// getColumnIdListFromPrimaryKeyRequest return list of column Id from PrimaryKeyRequest
+func getColumnIdListFromPrimaryKeyRequest(pkRequest PrimaryKeyRequest) []int {
+
+	cidlist := []int{}
+
+	for i := 0; i < len(pkRequest.Columns); i++ {
+		cidlist = append(cidlist, pkRequest.Columns[i].ColumnId)
+	}
+	return cidlist
+}
+
+// getColumnIdListOfSpannerTablePrimaryKey return list of column Id from spannerTable PrimaryKey
+func getColumnIdListOfSpannerTablePrimaryKey(spannerTable ddl.CreateTable) []int {
+	cidlist := []int{}
+
+	for i := 0; i < len(spannerTable.Pks); i++ {
+		cid := getColumnId(spannerTable, spannerTable.Pks[i].Col)
+		cidlist = append(cidlist, cid)
+	}
+	return cidlist
+}
+
+// getColumnIdListOfSpannerTable return list of column Id from spannerTable ColDefs
+func getColumnIdListOfSpannerTable(spannerTable ddl.CreateTable) []int {
+	cidlist := []int{}
+
+	for _, column := range spannerTable.ColDefs {
+		cidlist = append(cidlist, column.Id)
+	}
+	return cidlist
+}
+
+// isValidColumnIds checks columnId is already present in schema.
+func isValidColumnIds(pkRequest PrimaryKeyRequest, spannertable ddl.CreateTable) bool {
+
+	cidRequestList := getColumnIdListFromPrimaryKeyRequest(pkRequest)
+	cidSpannerTableList := getColumnIdListOfSpannerTable(spannertable)
+	leftjoin := difference(cidRequestList, cidSpannerTableList)
+
+	if len(leftjoin) > 0 {
+		return false
+	}
+	return true
+}
+
+// isValidColumnOrder make sure two primary key column can not have same order.
+func isValidColumnOrder(pkRequest PrimaryKeyRequest) bool {
+
+	list := []int{}
+
+	for i := 0; i < len(pkRequest.Columns); i++ {
+		list = append(list, pkRequest.Columns[i].Order)
+	}
+
+	if duplicateInArray(list) == -1 {
+		return false
+	}
+
+	return true
+}
+
+func duplicateInArray(element []int) int {
+	visited := make(map[int]bool, 0)
+	for i := 0; i < len(element); i++ {
+		if visited[element[i]] == true {
+			return element[i]
+		} else {
+			visited[element[i]] = true
+		}
+	}
+	return -1
+}
+
+// difference gives list of element that are only present in first list.
+func difference(listone, listtwo []int) []int {
```

Comment:
utility methods like this one and others in this file which is not specific to primary key should be moved to utils.go or some similar utility file